internal/oc/release.go

```diff
@@ -158,6 +158,9 @@ func (r *release) GetReleaseArchitecture(log logrus.FieldLogger, releaseImage st
 	}
 	imageInfoStr, err := execute(log, r.executer, pullSecret, cmd)
 	if err != nil {
+		if strings.Contains(err.Error(), "the image is a manifest list and contains multiple images") {
```

Comment:
Is there not a better way to check for this?
I don't like matching error messages. It feels very fragile.

Reply:
Sadly no, I have opened https://bugzilla.redhat.com/show_bug.cgi?id=2111537 because currently we can't even get a JSON output, i.e.

```
$ ./oc image info -o json quay.io/openshift-release-dev/ocp-release:4.11.0-0.nightly-multi-2022-07-19-022323
error: the image is a manifest list and contains multiple images - use --filter-by-os to select from:

  OS            DIGEST
  linux/amd64   sha256:cd6bd311612b3d4448457f5a68924216b087b937aa0f565cbfe4a8a42ecb6007
  linux/ppc64le sha256:6442ff2b7be3659424af383502006562c8a5753cefd4fc130626af98ad6d5946
  linux/s390x   sha256:ab0a0d47911b4776e7a1978cbc66ef80245b73f529f8b5f6ad29b729702c172b
  linux/arm64   sha256:e348c7a54f422eb3f1b5ad1cf21e332332b67327843099bc3b2ab62bc1114eff
```

Reply:
yikes:( a proper indication would indeed be nice. Seems there's an open task for it from a while ago: https://issues.redhat.com/browse/WRKLDS-222.
For now, might worth to use 'podman manifest inspect', which should list all platforms. Then we could have a list of all available archs with some jq magic:)

E.g.
```
podman manifest inspect quay.io/openshift-release-dev/ocp-release:4.11.0-0.nightly-multi-2022-07-19-022323

{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 1583,
            "digest": "sha256:cd6bd311612b3d4448457f5a68924216b087b937aa0f565cbfe4a8a42ecb6007",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 1583,
            "digest": "sha256:6442ff2b7be3659424af383502006562c8a5753cefd4fc130626af98ad6d5946",
            "platform": {
                "architecture": "ppc64le",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 1583,
            "digest": "sha256:ab0a0d47911b4776e7a1978cbc66ef80245b73f529f8b5f6ad29b729702c172b",
            "platform": {
                "architecture": "s390x",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
            "size": 1583,
            "digest": "sha256:e348c7a54f422eb3f1b5ad1cf21e332332b67327843099bc3b2ab62bc1114eff",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        }
    ]
}
```

Reply:
:laughing: I sent exactly that to @mkowalski earlier. It's the only solution I could come up with too.

Reply:
Yeah folks, this is very nice and elegant as I already discussed with Nick. But, there is only one but. This would mean that Assisted service container image needs to contain Podman (with all its dependencies) because in principle we are not allowed to talk to the container engine running on the cluster where assisted's Pod is placed.

So I'm very hesitant... It's a big dependency and in principle the very basic question people start asking is "why do you need podman inside a container?". I agree that it gives us an elegant way to deal with all this, I'm just not sure if we don't get immediately stopped when trying to introduce such a dependency.

Reply:
Right, seems that even using skopeo as suggested require quite a few dependencies. So need to decide if worth the effort at this point. We can also add some todo comment with a link to the oc bug instead:/

Reply:
Added a comment linking the Jira and BZ